### DISCUSSION TITLE
"Tagged All" interface is not member of any VLAN

### DISCUSSION THREAD

**Discussion by ZSasha0** [netbox-community/netbox#5702] [Q&A] [answered]
Hi guys,
I was playing with Netbox (v2.10.4) and it came to me that an interface with its 802.1Q Mode set as "Tagged (All)" is in fact not a member of ANY VLAN configured in NetBox.

Is it a bug or feature? 

I have a number of VLANs defined in NetBox. 
If I mark an interface as "Tagged" and then list some "Tagged VLANs" there , I am able to see this interface as a member of all VLANs I just specified.
However if I mark the interface as "Tagged All", that interface could no longer be seen as a member of **any** of those VLANs (in fact - it is not a member of any VLANs at all).

It makes things like "listing all interfaces which are members of VLAN X" impossible.
As a workaround, I can still make the interface to be "Tagged ALL" AND specify set of VLANs (via API calls) - in that case this interface is perfectly visible as a member of those VLANs, but the drawback of that that if I try to edit this interface later via GUI - all additional VLANs are removed.

If you are going to suggest "well, use only 'Tagged' with a list of VLANs and don't use 'Tagged All'" - the reason for me using 'Tagged All' is 1) it does not display hundreds or thousands of VLANs in GUI against each interface (as it does when an interface is "Tagged" only type) and b) I would assume that 'Tagged All' interface to become a member of any future added VLANs automatically.

Any thoughts/comments?
I am fairly new to NetBox so may be it is me doing something wrong?

Thank you.

**Comment by candlerb** [comment, marked as answer]
IMO, it's a feature.  "Tagged (All)" means that this port is tagged and has all VLANs that the switch knows about, without having to list them one by one.  Many switches have this capability, to set a trunk port without explicitly listing enabled VLANs.

However, this doesn't necessarily mean that the port is a member of all VLANs *that Netbox knows about* - only those VLANs which have been created on the switch.

Consider these cases:
* VLANs which are used in one site but not another
* VLANs which are used in one rack but not another
* Multiple VLANs with the same tag in the same site

Therefore, in general Netbox cannot associate *all* VLANs in its database (or even *all* VLANs for a given site) with a port marked "Tagged (All)"

-----

I imagine it would be possible to add a feature whereby you assign a list of VLANs at the *Device* level, as well as at the *Interface* level.  Then an interface marked "Tagged (All)" would be implicitly a member of all the VLANs which were configured on the Device.  However that would add complexity to the data model, and not really save much work if the device has only one or two trunk ports.

Another approach might be if Netbox could assign a VLAN Group to an interface (or device).  However, currently a given VLAN can only be a member of one VLAN group, which probably isn't flexible enough for this use case, as different devices may use different subsets of the same VLANs.

**Reply by ZSasha0** [reply]
Thank you @candlerb  , what you just said  totally makes sense. 
I kind of forgot that strange logic NB implemented in terms of VLANs. 
In all other world a VLAN belongs to a device but not here.
Anyway, thank you for clarifying that for me.
Cheers.
